File: bls_jztc/jztc/src/api/comment.ts
import request from '@/utils/request';

// 通用响应结构
export interface ApiResponse<T> {
  code: number;
  message: string;
  data: T;
}

// 评论项接口
export interface CommentItem {
  id: number;
  contentId: number;
  contentTitle: string;
  clientId: number;
  realName: string;
  comment: string;
  status: string;
  statusText: string;
  createdAt: string;
  updatedAt: string;
}

// 评论列表请求参数
export interface CommentListReq {
  page?: number;
  pageSize?: number;
  contentId?: number;
  status?: string;
  realName?: string;
  comment?: string;
}

// 评论列表响应
export interface CommentListRes {
  code: number;
  message: string;
  data: {
    list: CommentItem[];
    total: number;
    page: number;
  };
}

// 评论详情响应
export interface CommentDetailItem {
  id: number;
  contentId: number;
  contentTitle: string;
  clientId: number;
  realName: string;
  comment: string;
  status: string;
  createdAt: string;
  updatedAt: string;
}

// 评论详情响应
export interface CommentDetailRes {
  code: number;
  message: string;
  data: CommentDetailItem;
}

// 创建评论请求参数
export interface CommentCreateReq {
  contentId: number;
  clientId: number;
  realName: string;
  comment: string;
  status?: string;
}

// 创建评论响应
export interface CommentCreateRes {
  code: number;
  message: string;
  data: {
    id: number;
  };
}

// 更新评论请求参数
export interface CommentUpdateReq {
  id: number;
  comment: string;
  realName: string;
  status?: string;
}

// 更新评论响应
export interface CommentUpdateRes {
  code: number;
  message: string;
  data: null;
}

// 删除评论响应
export interface CommentDeleteRes {
  code: number;
  message: string;
  data: null;
}

// 更新评论状态请求参数
export interface CommentStatusUpdateReq {
  id: number;
  status: string;
  realName: string;
}

// 更新评论状态响应
export interface CommentStatusUpdateRes {
  code: number;
  message: string;
  data: null;
}

// 内容评论列表响应
export interface ContentCommentsRes {
  code: number;
  message: string;
  data: {
    list: CommentItem[];
    total: number;
    page: number;
  };
}

/**
 * 获取评论列表
 * @param params 查询参数
 * @returns 评论列表
 */
export const getCommentList = (params: CommentListReq): Promise<CommentListRes> => {
  return request.get('/comment/list', { params });
};

/**
 * 获取指定内容的评论列表
 * @param contentId 内容ID
 * @param params 分页参数
 * @returns 评论列表
 */
export const getContentComments = (contentId: number, params?: { page?: number; pageSize?: number }): Promise<ContentCommentsRes> => {
  return request.get('/comment/content-comments', { 
    params: { 
      contentId,
      ...params
    } 
  });
};

/**
 * 获取评论详情
 * @param id 评论ID
 * @returns 评论详情
 */
export const getCommentDetail = (id: number): Promise<CommentDetailRes> => {
  return request.get('/comment/detail', { params: { id } });
};

/**
 * 创建评论
 * @param data 评论数据
 * @returns 创建结果
 */
export const createComment = (data: CommentCreateReq): Promise<CommentCreateRes> => {
  return request({
    url: '/comment/create',
    method: 'post',
    data,
  });
};

/**
 * 更新评论
 * @param data 评论数据
 * @returns 更新结果
 */
export const updateComment = (data: CommentUpdateReq): Promise<CommentUpdateRes> => {
  return request({
    url: '/comment/update',
    method: 'put',
    data,
  });
};

/**
 * 删除评论
 * @param id 评论ID
 * @returns 删除结果
 */
export const deleteComment = (id: number): Promise<CommentDeleteRes> => {
  return request.delete('/comment/delete', { params: { id } });
};

/**
 * 更新评论状态
 * @param data 状态数据
 * @returns 更新结果
 */
export const updateCommentStatus = (data: CommentStatusUpdateReq): Promise<CommentStatusUpdateRes> => {
  return request({
    url: '/comment/status/update',
    method: 'put',
    data,
  });
}; 

File: bls_jztc/demo/api/comment/v1/comment.go
package v1

import (
	"github.com/gogf/gf/v2/frame/g"
)

// CommentListReq 评论列表请求
type CommentListReq struct {
	g.Meta    `path:"/list" method:"get" tags:"评论管理" summary:"获取评论列表" security:"Bearer" description:"获取评论列表，支持分页和搜索，需要管理员权限"`
	Page      int    `v:"min:1#页码最小值为1" json:"page" dc:"页码"`
	PageSize  int    `v:"max:100#每页最大100条" json:"pageSize" dc:"每页数量"`
	ContentId int    `json:"contentId" dc:"内容ID，不填则查询所有内容的评论"`
	Status    string `json:"status" dc:"状态：已审核、待审核、已拒绝，不填则查询所有状态"`
	RealName  string `json:"realName" dc:"真实姓名关键词"`
	Comment   string `json:"comment" dc:"评论内容关键词"`
}

// CommentListRes 评论列表响应
type CommentListRes struct {
	g.Meta `mime:"application/json" example:"json"`
	List   []CommentListItem `json:"list" dc:"评论列表"`
	Total  int               `json:"total" dc:"总数量"`
	Page   int               `json:"page" dc:"当前页码"`
}

// CommentListItem 评论列表项
type CommentListItem struct {
	Id           int    `json:"id" dc:"评论ID"`
	ContentId    int    `json:"contentId" dc:"内容ID"`
	ContentTitle string `json:"contentTitle" dc:"内容标题"`
	ClientId     int    `json:"clientId" dc:"客户ID"`
	RealName     string `json:"realName" dc:"真实姓名"`
	Comment      string `json:"comment" dc:"评论内容"`
	Status       string `json:"status" dc:"状态"`
	StatusText   string `json:"statusText" dc:"状态文本"`
	CreatedAt    string `json:"createdAt" dc:"创建时间"`
	UpdatedAt    string `json:"updatedAt" dc:"更新时间"`
}

// CommentDetailReq 评论详情请求
type CommentDetailReq struct {
	g.Meta `path:"/detail" method:"get" tags:"评论管理" summary:"获取评论详情" security:"Bearer" description:"获取评论详情，需要管理员权限"`
	Id     int `v:"required#评论ID不能为空" json:"id" dc:"评论ID"`
}

// CommentDetailRes 评论详情响应
type CommentDetailRes struct {
	g.Meta       `mime:"application/json" example:"json"`
	Id           int    `json:"id" dc:"评论ID"`
	ContentId    int    `json:"contentId" dc:"内容ID"`
	ContentTitle string `json:"contentTitle" dc:"内容标题"`
	ClientId     int    `json:"clientId" dc:"客户ID"`
	RealName     string `json:"realName" dc:"真实姓名"`
	Comment      string `json:"comment" dc:"评论内容"`
	Status       string `json:"status" dc:"状态"`
	CreatedAt    string `json:"createdAt" dc:"创建时间"`
	UpdatedAt    string `json:"updatedAt" dc:"更新时间"`
}

// ContentCommentsReq 内容评论列表请求
type ContentCommentsReq struct {
	g.Meta    `path:"/content-comments" method:"get" tags:"评论管理" summary:"获取指定内容的评论列表" security:"Bearer" description:"获取指定内容的评论列表，支持分页，需要管理员权限"`
	ContentId int `v:"required#内容ID不能为空" json:"contentId" dc:"内容ID"`
	Page      int `v:"min:1#页码最小值为1" json:"page" dc:"页码"`
	PageSize  int `v:"max:100#每页最大100条" json:"pageSize" dc:"每页数量"`
}

// ContentCommentsRes 内容评论列表响应
type ContentCommentsRes struct {
	g.Meta       `mime:"application/json" example:"json"`
	ContentId    int               `json:"contentId" dc:"内容ID"`
	ContentTitle string            `json:"contentTitle" dc:"内容标题"`
	List         []CommentListItem `json:"list" dc:"评论列表"`
	Total        int               `json:"total" dc:"总数量"`
	Page         int               `json:"page" dc:"当前页码"`
}

// CommentCreateReq 创建评论请求
type CommentCreateReq struct {
	g.Meta    `path:"/create" method:"post" tags:"评论管理" summary:"创建评论" security:"Bearer" description:"创建新评论，需要管理员权限"`
	ContentId int    `v:"required#内容ID不能为空" json:"contentId" dc:"内容ID"`
	ClientId  int    `v:"required#客户ID不能为空" json:"clientId" dc:"客户ID"`
	RealName  string `v:"required#真实姓名不能为空" json:"realName" dc:"真实姓名"`
	Comment   string `v:"required#评论内容不能为空" json:"comment" dc:"评论内容"`
	Status    string `v:"required|in:已审核,待审核,已拒绝#状态不能为空|状态值不正确" json:"status" dc:"状态：已审核、待审核、已拒绝"`
}

// CommentCreateRes 创建评论响应
type CommentCreateRes struct {
	g.Meta `mime:"application/json" example:"json"`
	Id     int `json:"id" dc:"评论ID"`
}

// CommentUpdateReq 更新评论请求
type CommentUpdateReq struct {
	g.Meta   `path:"/update" method:"put" tags:"评论管理" summary:"更新评论" security:"Bearer" description:"更新评论信息，需要管理员权限"`
	Id       int    `v:"required#评论ID不能为空" json:"id" dc:"评论ID"`
	RealName string `v:"required#真实姓名不能为空" json:"realName" dc:"真实姓名"`
	Comment  string `v:"required#评论内容不能为空" json:"comment" dc:"评论内容"`
	Status   string `v:"required|in:已审核,待审核,已拒绝#状态不能为空|状态值不正确" json:"status" dc:"状态：已审核、待审核、已拒绝"`
}

// CommentUpdateRes 更新评论响应
type CommentUpdateRes struct {
	g.Meta `mime:"application/json" example:"json"`
}

// CommentDeleteReq 删除评论请求
type CommentDeleteReq struct {
	g.Meta `path:"/delete" method:"delete" tags:"评论管理" summary:"删除评论" security:"Bearer" description:"删除评论，需要管理员权限"`
	Id     int `v:"required#评论ID不能为空" json:"id" dc:"评论ID"`
}

// CommentDeleteRes 删除评论响应
type CommentDeleteRes struct {
	g.Meta `mime:"application/json" example:"json"`
}

// CommentStatusUpdateReq 更新评论状态请求
type CommentStatusUpdateReq struct {
	g.Meta `path:"/status/update" method:"put" tags:"评论管理" summary:"更新评论状态" security:"Bearer" description:"更新评论状态，需要管理员权限"`
	Id     int    `v:"required#评论ID不能为空" json:"id" dc:"评论ID"`
	Status string `v:"required|in:已审核,待审核,已拒绝#状态不能为空|状态值不正确" json:"status" dc:"状态：已审核、待审核、已拒绝"`
}

// CommentStatusUpdateRes 更新评论状态响应
type CommentStatusUpdateRes struct {
	g.Meta `mime:"application/json" example:"json"`
}
